function-memoization/src/main/java/com/tomtom/kotlin/FunctionMemoizationExtensions.kt

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright 2020-present, TomTom N.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.tomtom.kotlin
+
+/**
+ * Creates a memoized function that will return cached result when the same inputs occurs again.
```

Comment:
return cached -> return a cached
occurs again -> are used again in a subsequent call